Ginger/Ginger/Environments/AppsListPage.xaml.cs

```diff
@@ -36,50 +39,65 @@ namespace Ginger.Environments
     /// </summary>
     public partial class AppsListPage : GingerUIPage
     {
-        public ProjEnvironment AppEnvironmnet { get; set; }
+        public ProjEnvironment AppEnvironment { get; set; }
         public AppsListPage(ProjEnvironment env)
         {
             InitializeComponent();
 
-            AppEnvironmnet = env;
-            CurrentItemToSave = AppEnvironmnet;
+            AppEnvironment = env;
+            CurrentItemToSave = AppEnvironment;
             //Set grid look and data
             SetGridView();
             SetGridData();
 
             BindingHandler.ObjFieldBinding(EnvNameTextBox, TextBox.TextProperty, env, ProjEnvironment.Fields.Name);
             EnvNameTextBox.AddValidationRule(new EnvironemntNameValidationRule());
-            xShowIDUC.Init(AppEnvironmnet);
-            BindingHandler.ObjFieldBinding(xPublishcheckbox, CheckBox.IsCheckedProperty, AppEnvironmnet, nameof(RepositoryItemBase.Publish));
+            xShowIDUC.Init(AppEnvironment);
+            BindingHandler.ObjFieldBinding(xPublishcheckbox, CheckBox.IsCheckedProperty, AppEnvironment, nameof(RepositoryItemBase.Publish));
 
             InitReleaseComboBox();
 
             grdApps.btnAdd.AddHandler(Button.ClickEvent, new RoutedEventHandler(AddApp));
             grdApps.AddToolbarTool("@Share_16x16.png", "Add Selected Applications to All Environments", new RoutedEventHandler(AddAppsToOtherEnvironments));
 
-            TagsViewer.Init(AppEnvironmnet.Tags);
+            TagsViewer.Init(AppEnvironment.Tags);
         }
 
         private void AddApp(object sender, RoutedEventArgs e)
         {
-            EnvApplication app = new EnvApplication();
-            app.Name = "New";
-            app.Active = true;
-            AppEnvironmnet.Applications.Add(app);
+            var ApplicationPlatforms = WorkSpace.Instance.Solution.ApplicationPlatforms.Where((app) => !AppEnvironment.CheckIfApplicationPlatformExists(app.Guid, app.AppName))?.ToList();
+
+
+
+            if (ApplicationPlatforms == null || ApplicationPlatforms?.Count == 0)
```

Comment:
ApplicationPlatforms?.Count == 0 keep only this condition as ? will be already checking the null condition

Reply:
Ok done